model-server/src/main/kotlin/org/modelix/model/server/handlers/KeyValueLikeModelServer.kt

```diff
@@ -298,7 +280,46 @@ class KeyValueLikeModelServer(val storeClient: IStoreClient) {
                 throw NotFoundException("Referenced key $key not found")
             }
         }
-        storeClient.putAll(newEntries)
+
+        // Entries were previously written directly to the store.
+        // Now we use the RepositoriesManager to merge changes instead of just overwriting a branch.
+
+        val hashedObjects = LinkedHashMap<String, String>()
+        val branchChanges = LinkedHashMap<BranchReference, String>()
+        val userDefinedEntries = LinkedHashMap<String, String?>()
+
+        for ((key, value) in newEntries) {
+            when {
+                HashUtil.isSha256(key) -> {
+                    hashedObjects[key] = value ?: throw IllegalArgumentException("No value provided for $key")
+                }
+                BranchReference.tryParseBranch(key) != null -> {
+                    branchChanges[BranchReference.tryParseBranch(key)!!] = value ?: throw IllegalArgumentException("Deleting branch $key not allowed")
```

Comment:
question: Could and should we delete the branch in v1 and propagate it to v2?

I have seen users of the v1 server deleting branches like this.

```java
final RestWebModelClient client = ....
client.getAsyncStore().put(repositoryId.getBranchReference("master").getKey(), null);
```

This new check would break it. Or am I missing something?